lib/pub_sub/message.rb

```diff
@@ -11,20 +11,22 @@ def initialize(payload)
     end
 
     def process
-      validate_message!
+      begin
+        validate_message!
+      rescue PubSub::ServiceUnknown => e
+        return
+      end
       handler.process(data)
     end
 
     def validate_message!
       messages = PubSub.config.subscriptions[sender]
-      if messages.nil?
-        warning = "WARN: We received a message from #{sender} but we do " \
-                'not subscribe to that service.'
-        PubSub.logger.warn(warning)
-      end
-      unless messages.include?(type)
-        error = "We received a message from #{sender} but it was " \
-                "of unknown type #{type}."
+      if messages.blank?
+        warning = "#{PubSub.config.service_name} received a message from #{sender} but no matching subscription exists for that sender"
```

Comment:
I should also add that I would not say the same if it we a chain of method calls. Here, we are dealing with a simple error message. There's no point in having it occupy multiple rows